src/js/references/components/Detail/ReferenceSettings.tsx

```diff
@@ -12,23 +14,59 @@ type ReferenceSettingsProps = {
     match: match<{ refId: string }>;
 };
 
+/**
+ * The reference settings view allowing users to manage the reference
+ */
 export default function ReferenceSettings({ match }: ReferenceSettingsProps) {
     const { refId } = match.params;
-    const { data, isLoading } = useGetReference(refId);
+    const [term, setTerm] = useState("");
+    const { data: reference, isLoading: isLoadingReference } = useGetReference(refId);
+    const {
+        data: users,
+        isLoading: isLoadingUsers,
+        isFetchingNextPage: isFetchingUsersNextPage,
+        fetchNextPage: fetchUsersNextPage,
+    } = useInfiniteFindUsers(25, term);
+    const {
+        data: groups,
+        isLoading: isLoadingGroups,
+        isFetchingNextPage: isFetchingGroupsNextPage,
+        fetchNextPage: fetchGroupsNextPage,
+    } = useInfiniteFindGroups(25, term);
 
-    if (isLoading) {
+    if (isLoadingReference || isLoadingUsers || isLoadingGroups) {
         return <LoadingPlaceholder />;
     }
 
     return (
         <>
-            {Boolean(data.remotes_from) || <LocalSourceTypes />}
+            {Boolean(reference.remotes_from) || <LocalSourceTypes />}
             <SectionHeader>
                 <h2>Access</h2>
                 <p>Manage who can access this reference.</p>
             </SectionHeader>
-            <ReferenceMembers noun="user" members={sortBy(data.users, "id")} refId={refId} />
-            <ReferenceMembers noun="group" members={sortBy(data.groups, "id")} refId={refId} />
+            <ReferenceMembers
+                data={users}
+                noun="user"
+                members={sortBy(reference.users, "id")}
+                refId={refId}
+                term={term}
+                setTerm={setTerm}
+                isFetchingNextPage={isFetchingUsersNextPage}
+                fetchNextPage={fetchUsersNextPage}
+                isLoading={isLoadingUsers}
+            />
+            <ReferenceMembers
+                data={groups}
+                noun="group"
+                members={sortBy(reference.groups, "id")}
+                refId={refId}
+                term={term}
+                setTerm={setTerm}
+                isFetchingNextPage={isFetchingGroupsNextPage}
+                fetchNextPage={fetchGroupsNextPage}
+                isLoading={isLoadingGroups}
```

Comment:
To me the state seems to far up. The data seems to only be used by components further down in the tree. I think it would be wroth taking the time to investigate a solution that localises data + state where it is getting used rather than so far up